// angular-cc-library/esm2015/lib/credit-card-directives.module.js
import { NgModule } from '@angular/core';
import { CreditCardFormatDirective } from './directives/credit-card-format.directive';
import { ExpiryFormatDirective } from './directives/expiry-format.directive';
import { CvcFormatDirective } from './directives/cvc-format.directive';
import * as ɵngcc0 from '@angular/core';
const CREDIT_CARD_LIBRARY_DIRECTIVES = [
    CreditCardFormatDirective,
    ExpiryFormatDirective,
    CvcFormatDirective,
];
export class CreditCardDirectivesModule {
}
CreditCardDirectivesModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: CreditCardDirectivesModule });
CreditCardDirectivesModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function CreditCardDirectivesModule_Factory(t) { return new (t || CreditCardDirectivesModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CreditCardDirectivesModule, { declarations: [CreditCardFormatDirective,
        ExpiryFormatDirective,
        CvcFormatDirective], exports: [CreditCardFormatDirective,
        ExpiryFormatDirective,
        CvcFormatDirective] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CreditCardDirectivesModule, [{
        type: NgModule,
        args: [{
                declarations: [CREDIT_CARD_LIBRARY_DIRECTIVES],
                exports: [CREDIT_CARD_LIBRARY_DIRECTIVES]
            }]
    }], null, null); })();
