// ngx-img-zoom/esm5/lib/ngx-img-zoom.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { NgxImgZoomComponent } from './ngx-img-zoom.component';
import { CommonModule } from '@angular/common';
import { NgxImgZoomService } from './ngx-img-zoom.service';
import { MouseWheelDirective } from './mouse-wheel.directive';
import * as ɵngcc0 from '@angular/core';
var NgxImgZoomModule = /** @class */ (function () {
    function NgxImgZoomModule() {
    }
NgxImgZoomModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxImgZoomModule });
NgxImgZoomModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgxImgZoomModule_Factory(t) { return new (t || NgxImgZoomModule)(); }, providers: [NgxImgZoomService], imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxImgZoomModule, { declarations: function () { return [NgxImgZoomComponent, MouseWheelDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [NgxImgZoomComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxImgZoomModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                providers: [NgxImgZoomService],
                declarations: [NgxImgZoomComponent, MouseWheelDirective],
                exports: [NgxImgZoomComponent]
            }]
    }], function () { return []; }, null); })();
    return NgxImgZoomModule;
}());
export { NgxImgZoomModule };
