// ngx-img-zoom/esm2015/lib/ngx-img-zoom.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class NgxImgZoomService {
    constructor() {
        this.zoomMode = 0 /* HoverZoom */;
    }
    // setZoomMode(zoomMode) {
    //   this.zoomMode = zoomMode;
    // }
    /**
     * @param {?} breakPoints
     * @return {?}
     */
    setZoomBreakPoints(breakPoints) {
        this.zoomBreakPoints = breakPoints;
    }
}
NgxImgZoomService.ɵfac = function NgxImgZoomService_Factory(t) { return new (t || NgxImgZoomService)(); };
NgxImgZoomService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: NgxImgZoomService, factory: NgxImgZoomService.ɵfac });
/** @nocollapse */
NgxImgZoomService.ctorParameters = () => [];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgxImgZoomService, [{
        type: Injectable
    }], function () { return []; }, null); })();
if (false) {
    /** @type {?} */
    NgxImgZoomService.prototype.zoomMode;
    /** @type {?} */
    NgxImgZoomService.prototype.zoomBreakPoints;
}
